### ISSUE TITLE
[WebGPU] device.importExternalTexture and queue.copyExternalImageToTexture are missing

### ISSUE THREAD

**Issue by alienself** [emscripten-core/emscripten#18190]
Hi,

I would like to recreate this [example](https://austin-eng.com/webgpu-samples/samples/texturedCube) and [this one](https://austin-eng.com/webgpu-samples/samples/videoUploading) using c++ and emscripten however the api call for device.importExternalTexture and queue.copyExternalImageToTexture are completely missing from the binding class.

This means that we currently cannot load up external images nor video texture within emscripten + webgpu :(

[https://austin-eng.com/webgpu-samples/samples/texturedCube](https://austin-eng.com/webgpu-samples/samples/texturedCube)
[https://austin-eng.com/webgpu-samples/samples/videoUploading](https://austin-eng.com/webgpu-samples/samples/videoUploading)

**Comment by jacobgorm** [comment]
Is this support still missing? I had a look at using https://github.com/juj/wasm_webgpu?tab=readme-ov-file instead to get this support, but that repo currently lacks Dawn compatibility on other platforms than WIN32.

**Comment by kainino0x** [comment]
Yes, it's still missing. We haven't dealt with this yet partly because there's no canonical way to refer to JS objects (HTMLVideoElement, VideoFrame, etc) from Wasm right now. We could use emval (which adds a dependency) or externref (which is not extremely well supported in emscripten yet).

I see there's code in wasm_webgpu for this but I would have to dig deeper to understand what they're actually doing.

If you need to call these functions, you should call them from JS (like via EM_ASM). It should be possible to use JsValStore to get the Device and other objects out from Wasm to JS:
https://github.com/emscripten-core/emscripten/blob/main/test/webgpu_jsvalstore.cpp
https://github.com/emscripten-core/emscripten/blob/main/src/library_html5_webgpu.js
Or you can do something like `WebGPU.mgrDevice.get(deviceId)`, though that api will probably change sooner than jsvalstore.

**Comment by kainino0x** [comment]
We're moving the WebGPU bindings to be maintained in Dawn instead of Emscripten, so I'm migrating this bug to Dawn here: https://crbug.com/413083271
